Add dark mode functionality

invaderb opened the pull request:
Add dark mode styling
Clean up svgs to make theme switching easier
Switch color select to use color picker
Refactor color picker to simplify logic

Tested on Chromium, Firefox, Safari, IE 11

![Screen Shot 2022-02-02 at 8 31 19 PM](https://user-images.githubusercontent.com/16693116/152276805-6ca37070-610a-4631-bb3f-db73704bc4f4.png)

invaderb committed:
Add dark mode functionality
Add dark mode styling
Clean up svgs to make theme switching easier
Switch color select to use color picker
Refactor color picker to simplify logic

NoHatCoder commented:
Why would you simplify the colour picker? The ability to choose chartreuse has specifically been enjoyed by the community. And not everyone knows how to use rgb values.

invaderb commented:
@NoHatCoder when I simplified it I also implemented the html color picker where you have a lot more control of the color and hue that you want.  A couple issues with using color names is there's only a limited selection of color names and browsers may not agree on that value should look like.  there is also a micro-optimization that rendering hex color values is faster than color names since it doesn't have invoke the lookup for that color name to the actual hex value.

![Screen Shot 2022-02-03 at 9 40 30 AM](https://user-images.githubusercontent.com/16693116/152390581-9e739665-2ce8-46da-83f9-d65d16a1644e.png)

NoHatCoder commented:
Oh, I missed the `type="color"` during inspection. So it is not really IE11 compatible, but it should fall back to just being a text field, so I guess we can use it.

Fyi CSS colours are standardized and well defined. And please don't assume that one thing is faster than another when you can't measure it and don't know how it is implemented.
